frame/support/procedural/src/pallet/expand/storage.rs

```diff
@@ -20,27 +20,66 @@ use crate::pallet::{
 	Def,
 };
 use frame_support_procedural_tools::clean_type_string;
-use std::collections::HashSet;
+use std::collections::HashMap;
 
-/// Generate the prefix_ident related the the storage.
+/// Generate the prefix_ident related to the storage.
 /// prefix_ident is used for the prefix struct to be given to storage as first generic param.
 fn prefix_ident(storage: &StorageDef) -> syn::Ident {
 	let storage_ident = &storage.ident;
 	syn::Ident::new(&format!("_GeneratedPrefixForStorage{}", storage_ident), storage_ident.span())
 }
 
+/// Generate the counter_prefix_ident related to the storage.
+/// counter_prefix_ident is used for the prefix struct to be given to counted storage map.
+fn counter_prefix_ident(storage_ident: &syn::Ident) -> syn::Ident {
+	syn::Ident::new(
+		&format!("_GeneratedCounterPrefixForStorage{}", storage_ident),
```

Comment:
I thought this is going to be called `format!("CounterFor{}", idnet)`. This is why we named the counters of staking `CounterFor` validators etc.

Reply:
We have a new counters in #9507, we should keep its name compatible with this PR so it can be migrated without any need for actual storage migration.

Reply:
the prefix used is for the counter of a storage is `"CounterFor" ++ storage_prefix`, this is just the name of the ident.

Reply:
ahh Okay 👍